### PR TITLE
OSDOCS-4782: Added HostNetwork note to Using the  network policy

### PR DESCRIPTION
**Pull request by dfitzmau**
Version(s):
4.12+

Issue:
[OSDOCS-4782](https://issues.redhat.com/browse/OSDOCS-4782)

Link to docs preview:
[Using the allow-from-hostnetwork network policy](https://80047--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/network_security/network_policy/about-network-policy.html#nw-networkpolicy-allow-from-hostnetwork_about-network-policy)

- [x] SME has approved this change. Ben Bennett
- [x] QE has approved this change.

### CODE REVIEW

File: modules/nw-networkpolicy-about.adoc [hunk 1]
```diff
@@ -150,11 +150,15 @@ spec:
...
 [id="nw-networkpolicy-allow-from-hostnetwork_{context}"]
-== Using the allow-from-hostnetwork network policy
+== Using the `allow-from-hostnetwork` network policy
+
+Add the following `allow-from-hostnetwork` `NetworkPolicy` object to direct traffic from the host network pods.
 
-Add the following `allow-from-hostnetwork` `NetworkPolicy` object to direct traffic from the host network pods:
+[IMPORTANT]
+====
+If you enabled `HostNetwork` endpoint publishing strategy for your cluster, pods in the `allow-from-host-network` namespace of the host node cannot communicate with pods from a different node in the same namespace.
```
Comment (asood-rh) [inline modules/nw-networkpolicy-about.adoc:160]: On the clusters that have endpoint publishing strategy 'LoadBalancer' the host networked pod on a node cannot communicate with pod on different  node in the same namespace.

Comment (dfitzmau) [reply modules/nw-networkpolicy-about.adoc]: :heavy_check_mark:

File: modules/nw-networkpolicy-about.adoc [hunk 2]
```diff
@@ -62,6 +62,11 @@ spec:
 
 * Only accept connections from pods within a project:
 +
+[IMPORTANT]
+====
+On an {product-title} cluster that uses the `LoadBalancerService` endpoint publishing strategy, the host network pod on a node cannot communicate with a pod on a different node in the same namespace. This configuration in combination with setting the `allow-from-same-namespace` network policy for your cluster would still result in the same communication limitation.
+====
```
Comment (npinaeva) [inline modules/nw-networkpolicy-about.adoc:68]: Talked to @asood-rh , and we agreed the following note may be more precise:
```
To allow ingress connections from hostNetwork pods in the same namespace, you need to apply "allow-from-hostnetwork" policy together with "allow-same-namespace"
```
Leaving it to the doc pros to rephrase it better :)

Comment (dfitzmau) [reply modules/nw-networkpolicy-about.adoc]: Thanks, @npinaeva . This note is good. Updated!

**Comment by michaelryanpeter** [comment]
lgtm

**Pull request merged**